Sources/HTTPTypes/HTTPFieldName.swift

```diff
@@ -48,6 +48,40 @@ extension HTTPField {
             self.canonicalName = name.lowercased()
         }
 
+        /// Create an HTTP field name from a string produced by HPACK or QPACK decoders used in
+        /// modern HTTP versions.
+        ///
+        /// - Warning: Do not use directly with the `HTTPFields` struct which does not allow pseudo
+        ///            header fields.
+        ///
+        /// - Parameter name: The name of the HTTP field or the HTTP pseudo header field. It must
+        ///                   be lowercased.
+        public init?(parsed name: String) {
+            guard !name.isEmpty else {
+                return nil
+            }
+            let token: Substring
+            if name.hasPrefix(":") {
+                token = name.dropFirst()
+            } else {
+                token = Substring(name)
+            }
+            guard token.utf8.allSatisfy({
+                switch $0 {
+                case 0x21, 0x23, 0x24, 0x25, 0x26, 0x27, 0x2A, 0x2B, 0x2D, 0x2E, 0x5E, 0x5F, 0x60, 0x7C, 0x7E:
```

Comment:
Should we define these somewhere so that it can match against human-readable names to reduce risk of missing something?

Reply:
You can use `UInt8(ascii: "")` for each one. It'll get a bit more verbose, mind.

Reply:
I can make a separate change to switch everything to that after landing this.